ddtrace/propagation/http.py

```diff
@@ -915,8 +963,19 @@ def _extract_configured_contexts_avail(normalized_headers):
     def _resolve_contexts(contexts, styles_w_ctx, normalized_headers):
         primary_context = contexts[0]
         links = []
+
         for context in contexts[1:]:
             style_w_ctx = styles_w_ctx[contexts.index(context)]
+
+            # special case where baggage is first in propagation styles
+            # why would you do that? I don't know, but it's possible
+            if style_w_ctx[0] == _PROPAGATION_STYLE_BAGGAGE:
+                baggage_context = contexts[0]
+                contexts.append(baggage_context)
+                del contexts[0]
+                styles_w_ctx.append(_PROPAGATION_STYLE_BAGGAGE)
+                del style_w_ctx[0]
```

Comment:
Can we handle this outside of the for-loop? baggage should never be the `primary_context`

Reply:
In the RFC it says that users can also disable other propagation styles (tracecontext or datadog) and enable only the baggage propagator which means baggage could be the `primar_context` in certain cases. Considering this, I think we can leave the code inside the for-loop. Let me know your thoughts because I think there are may be better ways to go about this.

Reply:
handle baggage separately entirely